### ISSUE TITLE
Error reporting in Epoch is broken

### ISSUE THREAD

**Issue by edwardalee** [lf-lang/lingua-franca#994]
Maybe due to recent changes in FileUtil and related classes, error reporting is broken in Epoch.  If I attempt to generate code for the following:
```
target C;
federated reactor {
    initial mode A { }
}
```
I get a null-pointer exception.  I should instead get this:
```
The currently selected code generation or target configuration does not support modal reactors!
```
(Modal models are not yet supported for federated reactors).

**Comment by cmnrd** [comment]
That is my bad! All this resource and iResource stuff is really confusing... I opened a fix PR.

**Comment by edwardalee** [comment]
Actually, it seems we aren't out of the woods yet. I'm still getting a NPE, but it seems impossible to tell where it is occurring because it is not caught.  Eclipse traps the error and displays the following:
<img width="1588" alt="image" src="https://user-images.githubusercontent.com/8513334/156608226-f62bf92a-4d9c-4b5c-865e-39f9ccf6dfc9.png">

Any ideas how to find out where the NPE is thrown?

**Comment by edwardalee** [comment]
Actually, still more problems. Epoch no longer marks compile errors.  Minimal example:
```
target C;
main reactor {
    reaction(startup) {=
        foo();
    =}
}
```
The console shows this error:
```
gmake: *** [Makefile:136: all] Error 2
ERROR: Users/eal/lingua-franca-master/git/lingua-franca/experimental/C/src/Junk.lf line 4
 error: implicit declaration of function 'foo' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
    /*Correspondence: Range: [(3, 8), (4, 4)) -> Range: [(0, 0), (1, 4)) (src=/resource/Experimental/src/Junk.lf)*/foo();
```
However, nothing is shown in Epoch:
![image](https://user-images.githubusercontent.com/8513334/156618912-3f42ff5f-f160-4907-84cf-d8d7f542ad70.png)

Note the complete lack of marks or error messages.

**Comment by cmnrd** [comment]
Again this was my bad. The `NullPointerException` in the error reporter was caused by the same change as the exception described in your original post. I pushed a fix to #998, with this change error reporting in Eclipse should work now as normal.

It is too bad that we cannot effectively test Epoch... I did a few smoke tests before merging #963, but apparently not the right ones.
